tests/dataframes_test.py

```diff
@@ -135,15 +138,112 @@ def test_count_and_sum_query(self):
+        if pipeline_dp.Metrics.SUM in metrics:
+            mock_aggregate.return_value = spark.sparkContext.parallelize([
+                ("key1", CountSumNamedTuple(1.2345, 2.34567))
+            ])
+        else:
+            mock_aggregate.return_value = spark.sparkContext.parallelize([
+                ("key1", CountNamedTuple(1.2345))
+            ])
+        df = spark.createDataFrame(get_pandas_df())
+        value_column = "value" if pipeline_dp.Metrics.SUM in metrics else None
+        columns = dataframes.Columns("privacy_key", "group_key", value_column)
+        metrics_dict = dict([(m, m.name) for m in metrics])
+        min_value = max_value = None
+        if pipeline_dp.Metrics.SUM in metrics:
+            min_value, max_value = -5, 5
+        bounds = dataframes.ContributionBounds(
+            max_partitions_contributed=5,
+            max_contributions_per_partition=2,
+            min_value=min_value,
+            max_value=max_value)
+        query = dataframes.Query(df, columns, metrics_dict, bounds, public_keys)
+
+        # Act
+        query.run_query(dataframes.Budget(1, 1e-10))
+
+        # Assert
+        expected_aggregate_params = pipeline_dp.AggregateParams(
+            max_partitions_contributed=5,
+            max_contributions_per_partition=2,
+            min_value=min_value,
+            max_value=max_value,
+            metrics=metrics)
+        mock_aggregate.assert_called_once_with(mock.ANY,
+                                               expected_aggregate_params,
+                                               mock.ANY,
+                                               public_partitions=public_keys)
+
+    def test_run_query_e2e_run(self):
+        # Arrange
+        spark = self._get_spark_session()
+        df = spark.createDataFrame(get_pandas_df())
+        columns = dataframes.Columns("privacy_key", "group_key", "value")
+        metrics = {
+            pipeline_dp.Metrics.COUNT: "count_column",
+            pipeline_dp.Metrics.SUM: "sum_column"
+        }
+        min_value = max_value = None
+        if pipeline_dp.Metrics.SUM in metrics:
+            min_value, max_value = -5, 5
+        bounds = dataframes.ContributionBounds(
+            max_partitions_contributed=2,
+            max_contributions_per_partition=2,
+            min_value=min_value,
+            max_value=max_value)
+        public_keys = ["key1", "key0"]
+        query = dataframes.Query(df, columns, metrics, bounds, public_keys)
+
+        # Act
+        budget = dataframes.Budget(1e6, 1e-1)  # large budget to get small noise
+        result_df = query.run_query(budget)
+
+        # Assert
+        pandas_df = result_df.toPandas()
+        pandas_df = pandas_df.sort_values(by=['group_key']).reset_index(
+            drop=True)
+        self.assertLen(pandas_df, 2)
+        # check row[0] = "key0", 0+noise, 0+noise
+        row0 = pandas_df.loc[0]
+        self.assertEqual(row0.group_key, "key0")
```

Comment:
why here we access it as a field and below as a dic (`row0["count_column"]`)? if both are possible should we use the same API everywhere?